src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
Let me be more specific on the second part of my comment. If we want to support lazy registration  i.e., register or initialize tsi implementation-specific (e.g., gts-specific) global resources if and only if the corresponding implementation is used in the user's code, the information will not be available at the time grpc_init() is called, and therefore, can not invoked correctly.